extensions/src/objectDetection/index.ts

```diff
@@ -0,0 +1,161 @@
+import { ArgumentType, BlockType, ExtensionMenuDisplayDetails, extension, block, untilTimePassed, rgbToHex, RGBObject } from "$common";
+import { ObjectDetector as ObjectDetectorClass } from "@mediapipe/tasks-vision"
+import { initializeObjectDetector, getImageHelper } from './utils'
+
+const details: ExtensionMenuDisplayDetails = {
+  name: "Object Detection",
+  description: "Detects and identifies the object shown!",
+  iconURL: "Typescript_logo.png",
+  insetIconURL: "typescript-logo.svg"
+};
+
+export default class objectDetection extends extension(details, "video", "drawable", "addCostumes", "toggleVideoBlock", "setTransparencyBlock") {
+
+  /**
+   * The MediaPipe detector that is used to detect objects
+   */
+  detector: ObjectDetectorClass;
+
+  /**
+   * Tells whether the continuous detection should be on/off
+   */
+  continuous: boolean;
+
+  /**
+   * Dimensions of the video frame
+   */
+  DIMENSIONS = [480, 360];
```

Comment:
This property should really be centralized onto the video mixin. Want to update that code to include this property? https://github.com/mitmedialab/prg-extension-boilerplate/blob/dev/extensions/src/common/extension/mixins/optional/video.ts

Reply:
Also, I think it'd be better to encode this as an object as 
```ts
{ width: number, height: number }
``` 
(instead of an array)